### PR TITLE
lint/refactor AttributeRow/Attributes

### PR DESCRIPTION
**Pull request by ngokevin**
- Use spread attributes to efficiently transfer props.
- Separate `getWidget` helper.
- Use `const`.

### CODE REVIEW

File: src/components/attributes/Attributes.js [hunk 1]
```diff
@@ -70,18 +69,43 @@ class AddComponent extends React.Component {
...
+  entity: React.PropTypes.element.isRequired
+};
 export default Attributes;
+
+/**
+ * Check if component has multiplicity.
+ */
+function isComponentInstanced (entity, componentName) {
+  for (var component in entity.components) {
+    if (component.substr(0,component.indexOf('__')) === componentName) {
```
Comment (cvan) [inline src/components/attributes/Attributes.js]: space after `,`

File: src/components/attributes/Attributes.js [hunk 2]
```diff
@@ -70,18 +69,43 @@ class AddComponent extends React.Component {
 }
 
 const Attributes = props => {
-  var entity = props.entity;
-  var components = entity ? props.entity.components : {};
+  const entity = props.entity;
+  const components = entity ? entity.components : {};
+  const defaultComponents = Object.keys(components).filter(function (key) {
+    return DEFAULT_COMPONENTS.indexOf(key) == -1;
```
Comment (cvan) [inline src/components/attributes/Attributes.js]: `===`

File: src/components/attributes/AttributeRow.js [hunk 3]
```diff
@@ -1,50 +1,72 @@
...
+      max: props.schema.hasOwnProperty('max') ? props.schema.max : Infinity
+    };
+
+    if (props.schema.oneOf && props.schema.oneOf.length > 0) {
+      return <SelectWidget {...widgetProps} options={props.schema.oneOf}/>;
+    }
+    if (type === 'map' || isMap) {
+      return <TextureWidget {...widgetProps}/>;
+    }
+    else if (type === 'number') {
```
Comment (cvan) [inline src/components/attributes/AttributeRow.js]: don't want a `switch`?

File: src/components/attributes/Attributes.js (hunk 1 above)
Comment (ngokevin) [inline src/components/attributes/Attributes.js]: Thanks, I just updated the linter to include all the semistandard rules.

File: src/components/attributes/Attributes.js [hunk 4]
```diff
@@ -1,66 +1,65 @@
...
-              <select ref="select">
-              {
-                Object.keys(AFRAME.components)
-                  .filter(function(key){return AFRAME.components[key].multiple || usedComponents.indexOf(key)==-1;})
-                  .sort()
-                  .map(function(value) {
-                    return <option key={value} value={value}>{value}</option>;
-                  }.bind(this))
-              }
+        <div className='collapsible-content'>
```
Comment (cvan) [inline src/components/attributes/Attributes.js:55]: using single quotation marks is preferred, I assume?

File: src/components/attributes/Attributes.js [hunk 5]
```diff
@@ -70,18 +69,43 @@ class AddComponent extends React.Component {
...
     </div>
   );
 };
+Attributes.PropTypes = {
+  entity: React.PropTypes.element.isRequired
+};
 export default Attributes;
+
+/**
+ * Check if component has multiplicity.
```
Comment (cvan) [inline src/components/attributes/Attributes.js:94]: `has multiplicity`? can we reword this?

File: src/components/attributes/Attributes.js (hunk 4 above)
Comment (ngokevin) [inline src/components/attributes/Attributes.js:55]: I used to use double. But the linter wants single, and makes sense to just go all the way single.

File: src/components/attributes/Attributes.js (hunk 5 above)
Comment (ngokevin) [inline src/components/attributes/Attributes.js:94]: I think it's fine in code context. At least it gives more context on top of the function name. Need to create core utils for these things anyways.

**Comment by fernandojsg** [comment]
Much cleaner, thanks!

**Pull request merged**